### PR TITLE
Implement automatic release

### PR DESCRIPTION
**Pull request by smoia**
As we discussed in our February developer meeting, we will adopt an automatic release to keep phy2bids updated in pipy, respecting the cold, unemotional, robotic rules of semantic versioning. 
 [Auto](https://github.com/intuit/auto) is a good tool for this: it operates using labels, rather than commit conventions such as angular, making it easier for everyone (especially less experienced contributors) to commit without fear. The idea is that every PR, if labelled correctly, will start the semantic versioning update.
Matched with a CI Pipy deployment, this should make our modifications be tracked at all times.

## Proposed Changes

  - [x] Implements [Auto](https://github.com/intuit/auto) for semantic versioning
  - [x] Implements a GH Action for automatic Pipy release.

### CODE REVIEW

**Comment by eurunuela** [comment]
This is awesome @smoia! It will be kind of hard to review I assume but it will clearly be worth it. 

Let me know if you need any help!

**Comment by smoia** [comment]
@rmarkello, I'm sorry to bother but I think I need your help setting this up. Could you have a look at travis and its configuration?

**Comment by rmarkello** [comment]
This is a shot in the dark, but can you try dropping from bionic back to xenial in the `.travis.yml`?

**Comment by smoia** [comment]
Mmm, unluckily that wasn't the issue. Any other idea?

**Comment by smoia** [comment]
Just trying to see the issue here

**Comment by smoia** [comment]
Ok, found it.
Apparently `.travis.yml` is very, _very_ peculiar on how things should be written.
There are few structures in each section that have to be maintaned:
- in `before_install`, it is not possible to have a long `if .. elif .. else` statement, but a series of `if` statements
- in `install`, it is important not to start the statements with a `-`
- in `script`, it is important to start the statement with a pipe (`|`).

I truly wonder why.

**Comment by smoia** [comment]
This is ready for review! However, contrarily to all the other PRs, please don't merge it once approved. I will do it when time is right!

File: docs/contributorfile.rst [hunk 1]
```diff
@@ -132,24 +146,23 @@ Issues labels
...
-Issues & PRs labels
-~~~~~~~~~~~~~~~~~~~
--  Documentation: Improvements or additions to documentation. This category includes (but is not limited to) docs pages, docstrings, and code comments.
--  Duplicate: Whatever this is, it exists already! Maybe it’s a closed Issue/PR, that should be reopened.
--  Enhancement: New features added or requested. This normally goes with a ``minormod`` label for PRs.
--  Outreach: As part of the scientific community, we care about outreach. Check the relevant section about it, but know that this Issue/PR contains information or tasks about abstracts, talks, demonstrations, papers.
--  Paused: Issue or PR should not be worked on until the resolution of other issues or PRs.
--  Testing: This is for testing features, writing tests or producing testing code. Both user testing and CI testing!
--  Urgent: If you don't know where to start, start here! This is probably related to a milestone due soon!
+-  BugFIX: These PRs close an issue labelled ``bug``. they also increase the semantic versioning for fixes (+0.0.1).
```
Comment (eurunuela) [inline docs/contributorfile.rst]:
```suggestion
-  BugFIX: These PRs close an issue labelled ``bug``. They also increase the semantic versioning for fixes (+0.0.1).
```

File: docs/contributorfile.rst [hunk 2]
```diff
@@ -132,24 +146,23 @@ Issues labels
...
+-  Internal: This PR contains changes to the internal API. It won't trigger a release, but it will be reported in the changelog.
+-  Documentation: See above. This PR won't trigger a release, but it will be reported in the changelog.
+-  Testing: See above. This PR won't trigger a release, but it will be reported in the changelog.
+-  Skip release: This PR will **not** trigger a release.
+-  Release: This PR will force the trigger of a release.
+
+Other labels
+^^^^^^^^^^^^
+-  Invalid: These PRs don't seem right. They actually seem so not right that they won’t be further processed. This label invalidates an Hacktoberfest contribution. If you think this is wrong, start a discussion in the relevant issue (or open one if missing). Reviewers are asked to give an explanation for the use of this label.
```
Comment (eurunuela) [inline docs/contributorfile.rst]:
```suggestion
-  Invalid: These PRs don't seem right. They actually seem so not right that they won’t be further processed. This label invalidates a Hacktoberfest contribution. If you think this is wrong, start a discussion in the relevant issue (or open one if missing). Reviewers are asked to give an explanation for the use of this label.
```

File: docs/contributorfile.rst [hunk 3]
```diff
@@ -132,24 +146,23 @@ Issues labels
 
 .. _prlabel:
 
-PRs labels
-~~~~~~~~~~
--  BugFIX: These PRs close an issue labelled ``bug``. they also increase the semantic versioning for fixes (+0.0.1).
--  Invalid: These PRs don't seem right. They actually seem so not right that they won’t be further processed. This label invalidates an Hacktoberfest contribution. If you think this is wrong, start a discussion in the relevant issue (or open one if missing). Reviewers are asked to give an explanation for the use of this label.
+PRs only labels
+~~~~~~~~~~~~~~~
+
```
Comment (eurunuela) [inline docs/contributorfile.rst]: Just a comment here. The spacing and weight of the titles feel a bit weird to me. I assume this is the result of the necessary break lines in rst.

File: .github/workflows/pythonpublish.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    runs-on: ubuntu-18.04
+
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+        python-version: 3.7
```
Comment (eurunuela) [inline .github/workflows/pythonpublish.yml]: Phys2bids supports Python 3.6. Could this be an issue?

Comment (smoia) [reply .github/workflows/pythonpublish.yml]: I don't know, actually. @rmarkello do you think we have to package for 3.6, 3.7, or both?

File: docs/contributorfile.rst (hunk 3 above)
Comment (smoia) [reply docs/contributorfile.rst]: Yep, it is!

File: .github/workflows/pythonpublish.yml (hunk 4 above)
Comment (rmarkello) [reply .github/workflows/pythonpublish.yml]: This is just installing the Python distro that will be used to bundle the package; it shouldn't limit which versions of Python versions can install the package, so we should be good!

File: docs/contributorfile.rst [hunk 5]
```diff
@@ -116,12 +116,26 @@ At physiopy, we use Issues and Milestones to keep track of and organise our work
 Labels
 ------
 The current list of labels are `here <https://github.com/physiopy/phys2bids/labels>`_. They can be used for **Issues**, **PRs**, or both.
+We use `auto <https://github.com/intuit/auto>`_ to automatise our semantic versioning and Pypi upload, so **it's extremely important to use the right PR labels**!
+
+.. _issueprlabels:
+
+Issues & PRs labels
```
Comment (rmarkello) [inline docs/contributorfile.rst]:
```suggestion
Issue & PR labels
```

File: docs/contributorfile.rst [hunk 6]
```diff
@@ -116,12 +116,26 @@ At physiopy, we use Issues and Milestones to keep track of and organise our work
...
+-  Outreach: As part of the scientific community, we care about outreach. Check the relevant section about it, but know that this Issue/PR contains information or tasks about abstracts, talks, demonstrations, papers.
+-  Paused: Issue or PR should not be worked on until the resolution of other issues or PRs.
+-  Testing: This is for testing features, writing tests or producing testing code. Both user testing and CI testing!
+-  Urgent: If you don't know where to start, start here! This is probably related to a milestone due soon!
 
 .. _issuelabel:
 
-Issues labels
-~~~~~~~~~~~~~
+Issues only labels
```
Comment (rmarkello) [inline docs/contributorfile.rst]:
```suggestion
Issue-only labels
```

File: docs/contributorfile.rst [hunk 7]
```diff
@@ -132,24 +146,23 @@ Issues labels
 
 .. _prlabel:
 
-PRs labels
-~~~~~~~~~~
--  BugFIX: These PRs close an issue labelled ``bug``. they also increase the semantic versioning for fixes (+0.0.1).
--  Invalid: These PRs don't seem right. They actually seem so not right that they won’t be further processed. This label invalidates an Hacktoberfest contribution. If you think this is wrong, start a discussion in the relevant issue (or open one if missing). Reviewers are asked to give an explanation for the use of this label.
+PRs only labels
```
Comment (rmarkello) [inline docs/contributorfile.rst]:
```suggestion
PR-only labels
```

**Comment by rmarkello** [comment]
Just wanted to check in and see if there was anything else to be done on this PR, @smoia ? It looks good to me&mdash;do you want to go ahead and merge it in?

**Comment by smoia** [comment]
I would like to see where #186 goes first.
If we split phy2bids, that is going to be a major release, and I would like to merge that before implementing auto.

**Comment by smoia** [comment]
Actually, the best thing would be to implement Auto _right_ before the major release.

However, there is something we need to do first: Auto creates the changelogs and the allcontributors automatically using the title and the labels of the PR. For this reason, we need to improve the PR titles so that it would make sense as a changelog point. If each one of us does their own PRs, this should be quick.

@vinferrer @eurunuela @rmarkello @RayStick @AyyagariA , could you check the titles of your PRs that  **are going to be merged or were merged after the 16th of February** (date of the latest release), please? Ideally they should start with a capital letter, describe the content of the Pr in a short sentence, and possibly start without any textual label (`[...]`).
Could you also check that the labels associated to the PRs reflect the contribution of the PR, please? If there is a `Documentation` or `Test` contribution associated to a `BugFIX`, `Minormod` or `Majormod`, please remove it/don't add it, but ping me in the issue.
When you checked your PRs, let me know with a message in this PR!

**Comment by RayStick** [comment]
@smoia - I checked my PRs since the date of latest release (which was **6th Feb** I think, not **16th Feb** as you wrote?). I just had two since then - #175 and #139. I changed their titles to be clearer, and to have a capital letter at the start. They each only have one label 'Documentation.'

**Comment by smoia** [comment]
Thank you @RayStick , and yes, you are right - it's **since the 6th of February** - thank you for pointing it out!

File: .autorc [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      "description": "Create a release when this pr is merged",
+      "releaseType": "release",
+      "overwrite": true
+    },
+    {
+      "name": "Internal",
+      "changelogTitle": "🏠 Internal",
+      "description": "Changes only affect the internal API",
+      "releaseType": "none",
+      "overwrite": true
```
Comment (vinferrer) [inline .autorc]: Why internal relases are not even a patch

**Comment by smoia** [comment]
This is how the changelog would look like in this moment, if auto started the release:

# v2.0.0 (Fri Mar 27 2020)

#### 💥 Breaking Change

- Stop reordering trigger [#156](https://github.com/physiopy/phys2bids/pull/156) ([@smoia](https://github.com/smoia))

#### 🚀 Enhancement

- Adjusts main function for easier testing and usage of phys2bids from inside python [#165](https://github.com/physiopy/phys2bids/pull/165) ([@eurunuela](https://github.com/eurunuela))
- Adds logger [#134](https://github.com/physiopy/phys2bids/pull/134) ([@eurunuela](https://github.com/eurunuela) [@vinferrer](https://github.com/vinferrer))

#### 🐛 Bug Fix

- Multifreq [#150](https://github.com/physiopy/phys2bids/pull/150) ([@vinferrer](https://github.com/vinferrer))
- [TEST] Consolidates / updates test suite [#180](https://github.com/physiopy/phys2bids/pull/180) ([@rmarkello](https://github.com/rmarkello))
- Trigger pictures improvement [#153](https://github.com/physiopy/phys2bids/pull/153) ([@vinferrer](https://github.com/vinferrer))
- Fix/logger [#176](https://github.com/physiopy/phys2bids/pull/176) ([@eurunuela](https://github.com/eurunuela))
- Adds integration tests [#166](https://github.com/physiopy/phys2bids/pull/166) ([@eurunuela](https://github.com/eurunuela))
- Add files for new logo & favicon [#170](https://github.com/physiopy/phys2bids/pull/170) ([@danalclop](https://github.com/danalclop))
- Trigger to 1 [#148](https://github.com/physiopy/phys2bids/pull/148) ([@vinferrer](https://github.com/vinferrer))
- correct outdir and name of multifreq files [#169](https://github.com/physiopy/phys2bids/pull/169) ([@vinferrer](https://github.com/vinferrer) [@smoia](https://github.com/smoia))
- fix frequency separation [#164](https://github.com/physiopy/phys2bids/pull/164) ([@vinferrer](https://github.com/vinferrer))
- [TEST] Speed-ups/improvements to test_txt.py [#178](https://github.com/physiopy/phys2bids/pull/178) ([@vinferrer](https://github.com/vinferrer))
- Change order of output folder existence check and creation before logger init [#154](https://github.com/physiopy/phys2bids/pull/154) ([@smoia](https://github.com/smoia))
- Corrected links [#160](https://github.com/physiopy/phys2bids/pull/160) ([@smoia](https://github.com/smoia))
- Doc/contributors [#159](https://github.com/physiopy/phys2bids/pull/159) ([@smoia](https://github.com/smoia))
- Added empty line in summary [#157](https://github.com/physiopy/phys2bids/pull/157) ([@smoia](https://github.com/smoia))
- solve list transpose that was buggy and problem with lines with extra blank tabs in reading [#152](https://github.com/physiopy/phys2bids/pull/152) ([@vinferrer](https://github.com/vinferrer))
- Moved first numeric line of AcqKnowledge files in txt format to the header, as it doesn't represents data [#147](https://github.com/physiopy/phys2bids/pull/147) ([@vinferrer](https://github.com/vinferrer))
- Started working on acq tests [#145](https://github.com/physiopy/phys2bids/pull/145) ([@eurunuela](https://github.com/eurunuela) [@smoia](https://github.com/smoia))
- Labchart no time [#140](https://github.com/physiopy/phys2bids/pull/140) ([@vinferrer](https://github.com/vinferrer))

#### ⚠️  Pushed to `master`

- Revert "removed eof" ([@smoia](https://github.com/smoia))
- Add tutorial_file json ([@smoia](https://github.com/smoia))
- Add json blueprint for file contribution, adjust list ([@smoia](https://github.com/smoia))
- Update howto.rst ([@smoia](https://github.com/smoia))
- Update contributorfile.rst ([@smoia](https://github.com/smoia))
- Removed pandas as dependency ([@smoia](https://github.com/smoia))
- Create .requires.yml ([@smoia](https://github.com/smoia))
- Update setup.cfg ([@smoia](https://github.com/smoia))
- Update requirements.txt ([@smoia](https://github.com/smoia))
- Added requires.io badge ([@smoia](https://github.com/smoia))
- Update .all-contributorsrc ([@smoia](https://github.com/smoia))
- Update README.md ([@smoia](https://github.com/smoia))
- Update index.rst ([@smoia](https://github.com/smoia))
- removed eof ([@smoia](https://github.com/smoia))
- Added anchor at the beginning of the page ([@smoia](https://github.com/smoia))
- Trying to fix Travis linting break ([@smoia](https://github.com/smoia))
- Update __init__.py ([@smoia](https://github.com/smoia))
- Update codecov.yml ([@smoia](https://github.com/smoia))
- Add __init__.py in interfaces ([@smoia](https://github.com/smoia))
- fixing images ([@smoia](https://github.com/smoia))
- removed images ([@smoia](https://github.com/smoia))
- corrected eol in gitattributes ([@smoia](https://github.com/smoia))
- Update and rename .codecov.yml to codecov.yml ([@smoia](https://github.com/smoia))
- Added forced eol to .gitattributes ([@smoia](https://github.com/smoia))

#### 🏠 Internal

- Upgrade to bionic [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Update docs/contributorfile.rst [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add script and autorc [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add install for auto [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Adding language and install for auto [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Remove sudo (deprecated) [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Create GH workflow to build and publish on release [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add new labels and badge to documentation [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Badging up! [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add badge template [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Remove sys [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Reset to master [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Corrected setup options [#149](https://github.com/physiopy/phys2bids/pull/149) ([@smoia](https://github.com/smoia))

#### 📝 Documentation

- Tiny addition to tutorial_file.json [#175](https://github.com/physiopy/phys2bids/pull/175) ([@RayStick](https://github.com/RayStick))
- how-to-use docs [#139](https://github.com/physiopy/phys2bids/pull/139) ([@RayStick](https://github.com/RayStick) [@smoia](https://github.com/smoia))
- Doc/contributors [#161](https://github.com/physiopy/phys2bids/pull/161) ([@smoia](https://github.com/smoia))
- Adds contributor guide to rtd [#158](https://github.com/physiopy/phys2bids/pull/158) ([@smoia](https://github.com/smoia))

#### Authors: 6

- Daniel Alcalá ([@danalclop](https://github.com/danalclop))
- Eneko Uruñuela ([@eurunuela](https://github.com/eurunuela))
- Rachael Stickland ([@RayStick](https://github.com/RayStick))
- Ross Markello ([@rmarkello](https://github.com/rmarkello))
- Stefano Moia ([@smoia](https://github.com/smoia))
- Vicente Ferrer ([@vinferrer](https://github.com/vinferrer))

**Comment by eurunuela** [comment]
That's super cool! 💯 

Also, the title of #145 is different from what is shown there. Any idea why? Just asking to make sure we're not missing anything.

**Comment by smoia** [comment]
Yes, I noticed it too. It seems that it doesn't use the updated PR title. I'm going to ask the developers.

**Comment by eurunuela** [comment]
We will never get the titles right from the start 🤣🤣 

Better if they can fix that (or maybe we could open a PR for them).

Are we sure they don’t have an option to get the updated titles?

**Comment by smoia** [comment]
I opened an issue, but I definitely don't have enough java nor GitHub API knowledge to know how to solve it.

Let's keep updating PR titles, merge this when ready, and if things will change I'll run auto to fix them.

File: .autorc (hunk 8 above)
Comment (hipstersmoothie) [reply .autorc]: `internal` is generally used for things like updating tests or anything project related that doesn't require a `patch`

File: .autorc [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    "all-contributors",
+    "conventional-commits",
+    "first-time-contributor",
+    "released"
+  ],
+  "release": {
+    "prerelease": true
+  },
+  "prereleaseBranches": [
+    "master"
```
Comment (hipstersmoothie) [inline .autorc]: you might be interested in [this](https://intuit.github.io/auto/pages/generated/shipit.html#without-%22next%22-branch-(--only-graduate-with-release-label)

Comment (smoia) [reply .autorc]: Thank you!
Wouldn't that unable us to automatically publish a release after every merged PR though?
From what I understand, that would trigger a new release only when the label `release` is added to the PR.
Our intentions are a bit different: we are an open development project, that wants to release as soon as there is something new, even if we're still in beta production stage.

The "prereleaseBranches" set on master is temporary, until we get out of beta stage.
Then we will remove it.

**Comment by smoia** [comment]
After adding a couple of tags, and following @hipstersmoothie bug fix (thank you!), this is how the changelog would look like:

# v2.0.0 (Fri Mar 27 2020)

#### 💥 Breaking Change

- Trigger is not reordered to channel 1 anymore [#156](https://github.com/physiopy/phys2bids/pull/156) ([@smoia](https://github.com/smoia))

#### 🚀 Enhancement

- Trigger pictures improvement [#153](https://github.com/physiopy/phys2bids/pull/153) ([@vinferrer](https://github.com/vinferrer))
- Adjusts main function for easier testing and usage of phys2bids from inside python [#165](https://github.com/physiopy/phys2bids/pull/165) ([@eurunuela](https://github.com/eurunuela))
- Multifreq [#150](https://github.com/physiopy/phys2bids/pull/150) ([@vinferrer](https://github.com/vinferrer))
- Adds logger functionality [#134](https://github.com/physiopy/phys2bids/pull/134) ([@eurunuela](https://github.com/eurunuela) [@vinferrer](https://github.com/vinferrer))

#### 🐛 Bug Fix

- Correct links [#160](https://github.com/physiopy/phys2bids/pull/160) ([@smoia](https://github.com/smoia))
- Fixes logger bug for Windows machines [#176](https://github.com/physiopy/phys2bids/pull/176) ([@eurunuela](https://github.com/eurunuela))
- correct outdir and name of multifreq files [#169](https://github.com/physiopy/phys2bids/pull/169) ([@vinferrer](https://github.com/vinferrer) [@smoia](https://github.com/smoia))
- fix frequency separation [#164](https://github.com/physiopy/phys2bids/pull/164) ([@vinferrer](https://github.com/vinferrer))
- Check existence and create output folder before logger initialisation [#154](https://github.com/physiopy/phys2bids/pull/154) ([@smoia](https://github.com/smoia))
- Trigger to 1 [#148](https://github.com/physiopy/phys2bids/pull/148) ([@vinferrer](https://github.com/vinferrer))
- Add contributors guidelines to documentation [#159](https://github.com/physiopy/phys2bids/pull/159) ([@smoia](https://github.com/smoia))
- Add empty line in summary [#157](https://github.com/physiopy/phys2bids/pull/157) ([@smoia](https://github.com/smoia))
- solve list transpose that was buggy and problem with lines with extra blank tabs in reading [#152](https://github.com/physiopy/phys2bids/pull/152) ([@vinferrer](https://github.com/vinferrer))
- put number of samples in the header instead of the channel_list [#147](https://github.com/physiopy/phys2bids/pull/147) ([@vinferrer](https://github.com/vinferrer))
- Labchart no time [#140](https://github.com/physiopy/phys2bids/pull/140) ([@vinferrer](https://github.com/vinferrer))

#### ⚠️  Pushed to `master`

- Revert "removed eof" ([@smoia](https://github.com/smoia))
- Add tutorial_file json ([@smoia](https://github.com/smoia))
- Add json blueprint for file contribution, adjust list ([@smoia](https://github.com/smoia))
- Update howto.rst ([@smoia](https://github.com/smoia))
- Update contributorfile.rst ([@smoia](https://github.com/smoia))
- Removed pandas as dependency ([@smoia](https://github.com/smoia))
- Create .requires.yml ([@smoia](https://github.com/smoia))
- Update setup.cfg ([@smoia](https://github.com/smoia))
- Update requirements.txt ([@smoia](https://github.com/smoia))
- Added requires.io badge ([@smoia](https://github.com/smoia))
- Update .all-contributorsrc ([@smoia](https://github.com/smoia))
- Update README.md ([@smoia](https://github.com/smoia))
- Update index.rst ([@smoia](https://github.com/smoia))
- removed eof ([@smoia](https://github.com/smoia))
- Added anchor at the beginning of the page ([@smoia](https://github.com/smoia))
- Trying to fix Travis linting break ([@smoia](https://github.com/smoia))
- Update __init__.py ([@smoia](https://github.com/smoia))
- Update codecov.yml ([@smoia](https://github.com/smoia))
- Add __init__.py in interfaces ([@smoia](https://github.com/smoia))
- fixing images ([@smoia](https://github.com/smoia))
- removed images ([@smoia](https://github.com/smoia))
- corrected eol in gitattributes ([@smoia](https://github.com/smoia))
- Update and rename .codecov.yml to codecov.yml ([@smoia](https://github.com/smoia))
- Added forced eol to .gitattributes ([@smoia](https://github.com/smoia))

#### 📝 Documentation

- Tiny edit to tutorial_file.json [#175](https://github.com/physiopy/phys2bids/pull/175) ([@RayStick](https://github.com/RayStick))
- Docs for how to use phys2bids [#139](https://github.com/physiopy/phys2bids/pull/139) ([@RayStick](https://github.com/RayStick) [@smoia](https://github.com/smoia))
- Add code of conduct to documentation [#161](https://github.com/physiopy/phys2bids/pull/161) ([@smoia](https://github.com/smoia))
- Add contributors guidelines to documentation [#158](https://github.com/physiopy/phys2bids/pull/158) ([@smoia](https://github.com/smoia))

#### ⚠️ Tests

- [TEST] Consolidates / updates test suite [#180](https://github.com/physiopy/phys2bids/pull/180) ([@rmarkello](https://github.com/rmarkello))
- Txt_test_faster [#178](https://github.com/physiopy/phys2bids/pull/178) ([@vinferrer](https://github.com/vinferrer))
- Adds integration tests [#166](https://github.com/physiopy/phys2bids/pull/166) ([@eurunuela](https://github.com/eurunuela))
- Adds acq tests [#145](https://github.com/physiopy/phys2bids/pull/145) ([@eurunuela](https://github.com/eurunuela) [@smoia](https://github.com/smoia))

#### 🏠 Internal

- Adding language and install for auto [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Added testing label [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add Outreach label [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Merge branch 'master' of github.com:smoia/phys2bids into auto [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Update docs/contributorfile.rst [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Create GH workflow to build and publish on release [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add script and autorc [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add install for auto [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Updated auto version [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Remove sudo (deprecated) [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Upgrade to bionic [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add new labels and badge to documentation [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Badging up! [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Add badge template [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Remove sys [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Reset to master [#181](https://github.com/physiopy/phys2bids/pull/181) ([@smoia](https://github.com/smoia))
- Correct setup options [#149](https://github.com/physiopy/phys2bids/pull/149) ([@smoia](https://github.com/smoia))

#### 🖋️  Outreach

- Add files for new logo & favicon [#170](https://github.com/physiopy/phys2bids/pull/170) ([@danalclop](https://github.com/danalclop))

#### Authors: 6

- Daniel Alcalá ([@danalclop](https://github.com/danalclop))
- Eneko Uruñuela ([@eurunuela](https://github.com/eurunuela))
- Rachael Stickland ([@RayStick](https://github.com/RayStick))
- Ross Markello ([@rmarkello](https://github.com/rmarkello))
- Stefano Moia ([@smoia](https://github.com/smoia))
- Vicente Ferrer ([@vinferrer](https://github.com/vinferrer))

**Comment by vinferrer** [comment]
Guys, I reviewed my PR and did the apropiate changes, can you confirm is fine?
imagen

**Comment by smoia** [comment]
@vinferrer the labels look ok (I'm going to add a `Skip release` on #148 due to the clash with later PRs).
However, if you could update the titles to be a bit more explicative, it would be nicer (especially #183 #178 #150 #147 #140)!
For instance, #183 could be something like "Add automatic method to count the number of MRI triggers". Let us know if you need more help in renaming PRs.

**Comment by vinferrer** [comment]
changes done

**Comment by eurunuela** [comment]
Hey @smoia! I see you're still working on this. When do you think we'll merge this?

**Comment by smoia** [comment]
@eurunuela, I'm planning to merge it in as soon as we close our [main milestone](https://github.com/physiopy/phys2bids/milestone/2), or at least #30, #36, #185 .

I'm sorry I worked on it again, but I just added a zenodo json file to configure its authorships automatically. Nothing much.

**Comment by smoia** [comment]
Ok. I'm going to merge this after updating the auto release version.

Finger crossed! :crossed_fingers: :crossed_fingers: :crossed_fingers:

**Comment by smoia** [comment]
By the way, @hipstersmoothie is there a way to check the latest auto release without using npm?
Also, do you want to have recognition for the review of this PR?

**Comment by hipstersmoothie** [comment]
There currently isn't a good way. Maybe `auto` could warn users of the binary there are updates.

**Pull request merged**